Добавление контекста

NikitaGoroshko7 opened the pull request:
https://trello.com/c/BRRCrdij/3-%D1%80%D0%B5%D0%B3%D0%B8%D1%81%D1%82%D1%80%D0%B0%D1%86%D0%B8%D1%8F-%D0%B8-%D0%B0%D0%B2%D1%82%D0%BE%D1%80%D0%B8%D0%B7%D0%B0%D1%86%D0%B8%D1%8F
Has been changed ApplicationContext
Has been added validation for models
Has been added CustomValidation
Has been added some new services in file startap and usings in file import.cs

NikitaGoroshko7 committed:
Добавление контекста
https://trello.com/c/mvTKxs64/1-%D0%B4%D0%BE%D0%B1%D0%B0%D0%B2%D0%BB%D0%B5%D0%BD%D0%B8%D0%B5-%D0%BA%D0%BE%D0%BD%D1%82%D0%B5%D0%BA%D1%81%D1%82%D0%B0
Has been edited register view and style for this
Added file import.cs for global usings
Has been deleted unuseful usings from files
Added EF Core and other tools for work with EF Core
Added migrations

GalayM reviewed (commented):
Need to use Identity context

NikitaGoroshko7 committed:
Has been added Identity context
Has been changed ApplicationContext
Has been added validation for models
Has been added CustomValidation
Has been added some new services and usings in file import.cs

GalayM commented on CrowdFundingApplication/Controllers/AccountController.cs:
```diff
@@ -1,14 +1,63 @@
+                }
+                else
+                    foreach (var error in result.Errors)
+                        ModelState.AddModelError(string.Empty, error.Description);
```
Always use {}

NikitaGoroshko7 replied:
Has been added {}

GalayM commented on CrowdFundingApplication/Controllers/AccountController.cs:
```diff
@@ -1,14 +1,63 @@
+                var result = await _signInManager.PasswordSignInAsync(model.Name, model.Password, model.RememberMe, false);
+
+                if (result.Succeeded)
+                    return RedirectToAction("MainPage", "Home");
```
Always use {}

NikitaGoroshko7 replied:
Has been added {}

GalayM commented on CrowdFundingApplication/Controllers/AccountController.cs:
```diff
@@ -1,14 +1,63 @@
+                if (result.Succeeded)
+                    return RedirectToAction("MainPage", "Home");
+                else
+                    ModelState.AddModelError("", "Неправильный логин и (или) пароль");
```
Always use {}

NikitaGoroshko7 replied:
Has been added {}

GalayM commented on CrowdFundingApplication/Controllers/HomeController.cs:
```diff
@@ -18,16 +9,25 @@ public HomeController(ILogger<HomeController> logger)
+        [Authorize]
+        public IActionResult MainPage()
+        {
+
```
Why is there empty space?

NikitaGoroshko7 replied:
Has been deleted the empty space

GalayM commented on CrowdFundingApplication/Data/ApplicationContext.cs:
```diff
@@ -0,0 +1,10 @@
+    {
+        Database.EnsureCreated();
+    }
+
```
???

NikitaGoroshko7 replied:
Has been deleted empty line

GalayM commented on CrowdFundingApplication/Models/InputViewModel.cs:
```diff
@@ -2,10 +2,14 @@
         public string Password { get; set; }
 
         public bool RememberMe { get; set; }
+
```
???

NikitaGoroshko7 replied:
Has been deleted empty line

GalayM commented on CrowdFundingApplication/Validation/CustomValidation.cs:
```diff
@@ -0,0 +1,20 @@
+                    Description = "Недопустимое имя пользоваля!"
+                });
+            }
+            return Task.FromResult(errors.Count == 0 ?
```
Any()

NikitaGoroshko7 replied:
Has been added Any() instend Count == 0

NikitaGoroshko7 committed:
Has been added changes in files
Has been added migrations
